pallets/account-manager/src/monetary.rs

```diff
@@ -0,0 +1,203 @@
+use sp_std::marker::PhantomData;
+
+use frame_support::{
+    pallet_prelude::*,
+    traits::{
+        tokens::{fungibles::Balanced, WithdrawConsequence},
+        Currency, ExistenceRequirement, ReservableCurrency, WithdrawReasons,
+    },
+};
+use sp_runtime::traits::Convert;
+
+pub struct Monetary<AccountId, Assets, NativeCurrency, AssetBalanceOf>(
+    PhantomData<(AccountId, Assets, NativeCurrency, AssetBalanceOf)>,
+);
+impl<
+        AccountId,
+        Assets: Balanced<AccountId>,
+        NativeCurrency: ReservableCurrency<AccountId>,
+        AssetBalanceOf: Convert<NativeCurrency::Balance, Assets::Balance>,
+    > Monetary<AccountId, Assets, NativeCurrency, AssetBalanceOf>
+{
+    pub fn deposit(
+        beneficiary: &AccountId,
+        asset_id: Option<Assets::AssetId>,
+        amount: NativeCurrency::Balance,
+    ) {
+        match asset_id {
+            None => {
+                NativeCurrency::deposit_creating(beneficiary, amount);
+            },
+            Some(asset_id) => {
+                Assets::deposit(asset_id, beneficiary, AssetBalanceOf::convert(amount)).unwrap();
```

Comment:
We shouldn't panic here, we should also make sure we prove the asset exists in or before the call to deposit and return an error if it doesn't or we cant.

Reply:
done